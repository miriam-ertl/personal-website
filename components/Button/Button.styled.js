import styled from "styled-components";

export const Button = styled.button`
  display: inline-flex;
  padding: 0.5rem 0.75rem;
  justify-content: center;
  align-items: center;
  gap: 0.625rem;
  border: ${(props) =>
    props.$secondary ? "1px solid var(--me-color-gray-700)" : "0"};
  background: ${(props) =>
    props.$secondary ? "var(--me-color-white)" : "var(--me-color-gray-700)"};
  color: ${(props) =>
    props.$secondary ? "var(--me-text-gray-700)" : "var(--me-text-inverted)"};
  text-align: center;
  font-size: 0.75rem;
  cursor: pointer;

  &:focus {
    color: ${(props) =>
      props.$secondary ? "var(--me-text-inverted)" : "var(--me-text-inverted)"};
    border: 2px solid var(--me-color-gray-200);
    background: var(--me-color-gray-500);
  }

  &:active {
    background: var(--me-color-gray-400);
    color: ${(props) =>
      props.$secondary ? "var(--me-text-inverted)" : "var(--me-text-inverted)"};
  }
`;
