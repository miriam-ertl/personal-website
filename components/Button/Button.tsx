import { Button } from "@/components/Button/Button.styled";

export default function StyledButton() {
  return <Button $secondary />;
}
