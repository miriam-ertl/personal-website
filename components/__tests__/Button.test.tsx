import { render, screen } from "@testing-library/react";

import Button from "@/components/Button/Button";

describe(Button.name, () => {
  it("renders a button", () => {
    render(<Button />);

    expect(screen.getByRole("button")).toBeInTheDocument();
  });
  it("renders button", () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });
});
