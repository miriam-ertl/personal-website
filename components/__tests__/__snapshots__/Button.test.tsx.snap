// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button 1`] = `
<div>
  <button
    class="sc-beySPh hfXiWF"
  />
</div>
`;

exports[`StyledButton renders button 1`] = `
<div>
  <button
    class="sc-beySPh hfXiWF"
  />
</div>
`;
